### PR TITLE
feat: add Prefab provider

### PR DESCRIPTION
**Pull request by liran2000**
[Readme](https://github.com/open-feature/go-sdk-contrib/blob/a5035bfaddb0f95a9d2e24fcfa9212fc4f30739d/providers/prefab/README.md#unofficial-prefab-openfeature-provider-for-go) describes the provider.

### CODE REVIEW

File: providers/prefab/README.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# Unofficial Prefab OpenFeature Provider for GO
```
Comment (liran2000) [inline providers/prefab/README.md:1]: hi @semanticart @jkebinger,

you are welcome to review.
this PR is similar to [Java provider PR](https://github.com/open-feature/java-sdk-contrib/pull/915).

Comment (liran2000) [reply providers/prefab/README.md:1]: Hi @semanticart
Will you be able to review this from Prefab point of view?

Comment (semanticart) [reply providers/prefab/README.md:1]: @liran2000 yes, I can take a look soon. Thanks!

File: providers/prefab/README.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+### Usage Example
+
+```go
+import (
+  provider "github.com/open-feature/go-sdk-contrib/providers/prefab/pkg"
+  of "github.com/open-feature/go-sdk/openfeature"
+  prefab "github.com/prefab-cloud/prefab-cloud-go/pkg"
+)
+
+of.SetProvider(provider)
```
Comment (semanticart) [inline providers/prefab/README.md:26]: I'm not sure if this is a bug or if the README omits the full code intentionally, but I found it a little confusing that I couldn't just copy directly from the README and have it work. 

Here's what did work for me:

```go
	config := provider.ProviderConfig{
		APIKey: "YOUR_API_KEY_GOES_HERE",
	}

	prefab, err := provider.NewProvider(config)
	if err != nil {
		fmt.Println(err)
		return
	}

	of.SetProvider(prefab)
```

**Review by semanticart — APPROVED** [review]
@liran2000 I left a couple notes but overall this LGTM and (once I got the provider set up properly) works as intended.

Thanks for your efforts here! 💙

File: providers/prefab/pkg/provider.go [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+			},
+		}
+	}
+
+	value, _ := p.PrefabClient.GetBoolValueWithDefault(flag, prefabContext, defaultValue)
+	if err == nil {
+		return of.BoolResolutionDetail{
+			Value: value,
+		}
+	}
```
[the remark this answers is not in the document]
Comment (liran2000) [reply providers/prefab/pkg/provider.go]: updated

File: providers/prefab/README.md (hunk 2 above)
Comment (liran2000) [reply providers/prefab/README.md:45]: I updated the readme, it should now work as is.

File: providers/prefab/README.md (hunk 1 above)
Comment (liran2000) [reply providers/prefab/README.md:1]: thanks.
@Kavindu-Dodan / @toddbaert you are welcome to review too.

File: providers/prefab/pkg/provider_test.go [hunk 4]
```diff
@@ -0,0 +1,390 @@
...
+	})
+
+	// TODO
+	// t.Run("sample_json", func(t *testing.T) {
+	// 	value, err := ofClient.ObjectValueDetails(context.Background(), "sample_json", map[string]interface{}{
+	// 		"nested": "value",
+	// 	}, evalCtx)
+	// 	require.Equal(t, []string{"a", "b"}, value.Value)
+	// 	require.Nil(t, err)
+	// })
```
Comment (toddbaert) [inline providers/prefab/pkg/provider_test.go:188]: Should this be removed?

File: providers/prefab/pkg/provider_config.go [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+
+type ProviderConfig struct {
+	Configs map[string]interface{}
+	APIKey  string
+	APIURLs []string
+	Sources []string
+	// EnvironmentNames             []string
+	// ProjectEnvID                 int64
+	// InitializationTimeoutSeconds float64
+	// OnInitializationFailure      OnInitializationFailure
```
Comment (toddbaert) [inline providers/prefab/pkg/provider_config.go:11]: Should this be removed or is there a reason to keep it? Perhaps comment a TODO if so?

File: providers/prefab/README.md [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+slice, _ := ofClient.ObjectValueDetails(context.Background(), "sample_list", []string{"a2", "b2"}, evalCtx)
+fmt.Printf("slice: %v\n", slice)
+
+of.Shutdown()
+
+```
+See [provider_test.go](./pkg/provider_test.go) for more information.
+
+## Notes
+Some Prefab custom operations are supported from the Prefab client via PrefabClient.
```
Comment (toddbaert) [inline providers/prefab/README.md:69]:
```suggestion

Some Prefab custom operations are supported from the Prefab client via PrefabClient.
```

File: providers/prefab/pkg/provider_test.go (hunk 4 above)
Comment (toddbaert) [reply providers/prefab/pkg/provider_test.go:188]: Oh I guess this is related to https://github.com/open-feature/go-sdk-contrib/pull/575/files#diff-cda080e5d119928854541dc260e063a153933ee2c950795929524c75e1655eb0R74

**Review by toddbaert — APPROVED** [review]
Can you add a component_owners entry for this with yourself and @semanticart if they agree?

Approved otherwise. Thanks!

**Review by Kavindu-Dodan — APPROVED** [review]

File: providers/prefab/pkg/provider_test.go (hunk 4 above)
Comment (liran2000) [reply providers/prefab/pkg/provider_test.go:188]: yes, can leave it for now, and will track to see when can be enhanced with the additional tests.

**Comment by liran2000** [comment]
> Can you add a component_owners entry for this with yourself and @semanticart if they agree?
> 
> Approved otherwise. Thanks!

@semanticart added you as well :) thanks !

**Pull request merged**